### PR TITLE
Update gas calculation in eth method

### LINKED ISSUE #118: Update gas calculation in eth method
### Problem

Currently all the gas related items use an estimated static value since mIrror Node had not yet implemented the fee retrieval logic.

As such the gas prices used are not precise

### Solution

Update the logic to consume the Mirror Node endpoint

- Add a `getNetworkFees()` to `mirrorNodeClient`
- Call method from `eth` methods to get the gas fee
- Updated `eth.spec.ts` with tests

### Alternatives

if Mirror Node endpoint is still not available add middle of sprint then consensusNode may be consulted
- Add a  `getNetworkFees()` to `sdkClient`, parsing out gas price
- Call method from `eth` methods to get the gas fee
- Updated `eth.spec.ts` with tests

In this manor `eth` should be oblivious to the implementation of the source retrieval.
This in a future sprint the mirrorNode may be consulted for information

### PR DESCRIPTION
**Pull request by ar-conmit**
Signed-off-by: Anton Rusev <anton.rusev.code@gmail.com>

**Description**:
- add `getNetworkFees` to retrieve fees from mirror-node
- update `getFeeWeibars` to use `getNetworkFees` and convert them in weibars
- update `eth_gasPrice` to return the actual gas price in weibars
- add cache functionallity to `eth_gasPrice`

**Related issue(s)**:

Fixes #118 

**Notes for reviewer**:
Problem - `Mocha` tests are hanging out when there is an unfinished process like `setTimeout`(used by `js-cache` module).
Solution - Added `--exit` flag to `test` script in `/relay/package.json` file to force `mocha` to quit right after the tests are completed

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -124,16 +128,22 @@ export class EthImpl implements Eth {
...
+    const networkFees = await this.mirrorNodeClient.getNetworkFees();
+
+    if (networkFees && Array.isArray(networkFees.fees)) {
+      const txFee = networkFees.fees.find(({transaction_type}) => transaction_type === "EthereumTransaction");
+      if (txFee && txFee.gas) {
+        // convert tinyBars into weiBars 
+        const weibars = Hbar
+          .fromTinybars(txFee.gas)
+          .toTinybars()
+          .multiply(10_000_000_000);
```
Comment (Daniel-K-Ivanov) [inline packages/relay/src/lib/eth.ts:140]: I assume that we have this constant used somewhere else. We can declare it as variable

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

File: packages/relay/src/lib/constants.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+enum CACHE_TTL {
+    ONE_HOUR = 3_600_000
+};
+
+export default {
+    TINYBAR_TO_WEIBAR_COEF: 10_000_000_000,
+    EXCHANGE_RATE_FILE_ID: "0.0.112",
+    CACHE_KEY,
+    CACHE_TTL,
+};
```
Comment (Nana-EC) [inline packages/relay/src/lib/constants.ts:33]: nit: new line

File: packages/relay/src/lib/constants.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+enum CACHE_KEY {
+    GAS_PRICE = 'gas_price'
+};
+enum CACHE_TTL {
+    ONE_HOUR = 3_600_000
+};
+
+export default {
+    TINYBAR_TO_WEIBAR_COEF: 10_000_000_000,
+    EXCHANGE_RATE_FILE_ID: "0.0.112",
```
Comment (Nana-EC) [inline packages/relay/src/lib/constants.ts:30]: nit:
```suggestion
    EXCHANGE_RATE_FILE_ID: '0.0.112',
```

Comment (Nana-EC) [inline packages/relay/src/lib/constants.ts:30]: Q: I don't see this used. Did you mean to consume it in `MirrorNodeClient`?
Can remove if not used

File: packages/relay/src/lib/eth.ts [hunk 4]
```diff
@@ -124,16 +126,22 @@ export class EthImpl implements Eth {
...
-
-    //contractTransactionGas is in tinycents * 1000, so the final multiplier is truncated by 3 zeroes for
-    // the conversion to weibars
-    return Math.ceil(
-      (contractTransactionGas / exchangeRates.currentRate.cents) * exchangeRates.currentRate.hbars * 10_000_000
-    );
+    const networkFees = await this.mirrorNodeClient.getNetworkFees();
+
+    if (networkFees && Array.isArray(networkFees.fees)) {
+      const txFee = networkFees.fees.find(({transaction_type}) => transaction_type === "EthereumTransaction");
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:132]: Make `EthereumTransaction` a static field.
Or add to constants if it is used elsewhere

**Review by mmrt — APPROVED** [review]

**Comment by Nana-EC** [comment]
So even with the new API mirror node returns 404 in a new env since there are no exchange rate records submitted.
We need to add a backup flow whereby if MirrorNode returns null we consult the network.

File: packages/relay/src/lib/eth.ts [hunk 5]
```diff
@@ -125,16 +130,43 @@ export class EthImpl implements Eth {
...
 
   private async getFeeWeibars() {
-    const exchangeRates = await this.sdkClient.getExchangeRate();
+    let networkFees = await this.mirrorNodeClient.getNetworkFees();
 
-    //FIXME retrieve fee from fee API when released
-    const contractTransactionGas = 853454;
+    if (_.isNil(networkFees)) {
+      this.logger.debug(`Mirror Node returned no fees. Fallback to network`);
+      const feeSchedules = await this.sdkClient.getFeeSchedule();
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:137]: Coverage for this will have to come with future integration tests that use `sinon` to mock `SDKClient class`

**Review by Nana-EC — APPROVED** [review]
LG

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**